### PR TITLE
[RM Ch07.9] IaaC Security Requirements

### PR DESCRIPTION
**Pull request by walterkozlowski**
This PR needs to be reviewed in conjunction with PR#2427, which provides an Intro contents for these Infrastructure as a Code Requirements.

### CODE REVIEW

**Review by ulikleber — APPROVED** [review]

**Review by bfcohen100 — APPROVED** [review]
Very important addition.

**Review by bfcohen — APPROVED** [review]
Very much needed.

**Comment by walterkozlowski** [comment]
@karinesevilla, @pgoyal01 - any comments, suggestions?  Thanks.

**Review by karinesevilla — APPROVED** [review]

File: doc/ref_model/chapters/chapter07.md [hunk 1]
```diff
@@ -560,6 +565,72 @@ The Platform is assumed to provide configurable alerting and notification capabi
...
+
+<p align="center"><b>Table 7-9:</b> IaaC - Secure Design and Architecture Stage Requirements</p>
+
+<a name="7.9.10"></a>
+### 7.9.10. IaaC - Secure Code Stage Requirements
+
+
+| Ref | Requirement | Definition/Note |
+|---|----|---|
+| req.sec.code.001 | SAST -Static Application Security Testing **should** be applied during Secure Coding stage triggered by Pull, Clone or Comment trigger. | Security testing that analyses application source code for software vulnerabilities and gaps against best practices. Example: open source OWASP range of tools.|
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.md:585]:
```suggestion
| req.sec.code.001 | SAST -Static Application Security Testing **must** be applied during Secure Coding stage triggered by Pull, Clone or Comment trigger. | Security testing that analyses application source code for software vulnerabilities and gaps against best practices. Example: open source OWASP range of tools.|
```

File: doc/ref_model/chapters/chapter07.md [hunk 2]
```diff
@@ -560,6 +565,72 @@ The Platform is assumed to provide configurable alerting and notification capabi
...
+| req.sec.code.002 | SCA – Software Composition Analysis **should** be applied during Secure Coding stage triggered by Pull, Clone or Comment trigger. | Security testing that analyses application source code or compiled code for software components with known vulnerabilities. Example: open source OWASP range of tools.  |
+| req.sec.code.003 | Source Code Review **should** be performed continuously during Secure Coding stage. | Typically done manually.  |
+| req.sec.code.004 | Integrated SAST via IDE Plugins **should** be used during Secure Coding stage triggered by Developer Code trigger. | On the local machine: through the IDE or integrated test suites; triggered on completion of coding be developer. |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.md:588]: This is an implementation of req.sec.code.001 -- does it belong in RM?

File: doc/ref_model/chapters/chapter07.md [hunk 3]
```diff
@@ -560,6 +565,72 @@ The Platform is assumed to provide configurable alerting and notification capabi
...
+| req.sec.code.005 | ISAST of Source Code Repo **should** be used during Secure Coding stage triggered by Developer Code trigger. | Continuous delivery pre-deployment: scanning prior to deployment. |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.md:589]: Is this requirement about Static testing of the repo? If yes, then suggest
```suggestion
| req.sec.code.005 | SAST of Source Code Repo **should** be performed during Secure Coding stage triggered by Developer Code trigger. | Continuous delivery pre-deployment: scanning prior to deployment. |
```

File: doc/ref_model/chapters/chapter07.md [hunk 4]
```diff
@@ -560,6 +565,72 @@ The Platform is assumed to provide configurable alerting and notification capabi
...
+
+
+| Ref | Requirement | Definition/Note |
+|---|----|---|
+| req.sec.bld.001 | SAST -Static Application Security Testing **should** be applied during the Continuous Build, Integration and Testing stage triggered by Build and Integrate trigger. | Example: open source OWASP range of tools.|
+| req.sec.bld.002 | SCA – Software Composition Analysis **should** be applied during the Continuous Build, Integration and Testing stage triggered by Build and Integrate trigger. | Example: open source OWASP range of tools.  |
+| req.sec.bld.003 | Container and Image Scan **must** be applied during the Continuous Build, Integration and Testing stage triggered by Package trigger. | Example: A push of a container image to a container registry may trigger a vulnerability scan before the image becomes available in the registry.  |
+| req.sec.bld.004 | DAST – Dynamic Application Security Testing **must** be applied during the Continuous Build, Integration and Testing stage triggered by Stage & Test trigger. | Security testing that analyses a running application by exercising application functionality and detecting vulnerabilities based on application behaviour and response. Example: OWASP ZAP. |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.md:602]:
```suggestion
| req.sec.bld.004 | DAST – Dynamic Application Security Testing **should** be applied during the Continuous Build, Integration and Testing stage triggered by Stage & Test trigger. | Security testing that analyses a running application by exercising application functionality and detecting vulnerabilities based on application behaviour and response. Example: OWASP ZAP. |
```

**Pull request merged**